admin/services/data-access/result.data.service.js

```diff
@@ -14,32 +64,36 @@ const R = require('ramda')
 resultDataService.sqlFindResultsBySchool = async (schoolId, checkWindowId) => {
   const sql = `
     SELECT
-    p.foreName,
-    p.lastName,
-    p.middleNames,
-    p.dateOfBirth,
-    pg.group_id,
-    latestPupilCheck.mark,
-    ac.reason
-    FROM ${sqlService.adminSchema}.pupil p
+        p.id,
+        latestPupilCheck.mark,
+        latestPupilCheck.maxMark,
+        cs.code as checkStatusCode,
+        ac.reason
+    FROM [mtc_admin].pupil p
     LEFT OUTER JOIN
-    (SELECT chk.pupil_id, chk.mark, ROW_NUMBER() OVER ( PARTITION BY chk.pupil_id ORDER BY chk.markedAt DESC ) as rank
-      FROM ${sqlService.adminSchema}.[check] chk
-      INNER JOIN ${sqlService.adminSchema}.checkStatus cs ON cs.id = chk.checkStatus_id
-      WHERE cs.code = 'CMP'
-      AND chk.checkWindow_id = @checkWindowId
-    ) latestPupilCheck
-      ON p.id = latestPupilCheck.pupil_id
-    LEFT JOIN ${sqlService.adminSchema}.pupilAttendance pa
-      ON (p.id = pa.pupil_id AND pa.isDeleted = 0)
-    LEFT JOIN ${sqlService.adminSchema}.attendanceCode ac
-      ON pa.attendanceCode_id = ac.id
-    LEFT OUTER JOIN ${sqlService.adminSchema}.[pupilGroup] pg
-      ON pg.pupil_id = p.id 
+        (SELECT
+          chk.pupil_id,
+          chk.checkStatus_id,
+          chk.mark,
+          chk.maxMark,
+          ROW_NUMBER() OVER ( PARTITION BY chk.pupil_id ORDER BY chk.id DESC ) as rank
+        FROM [mtc_admin].[check] chk
+        INNER JOIN [mtc_admin].checkStatus cs
+          ON cs.id = chk.checkStatus_id
+          AND cs.code IN ('NTR', 'CMP')
+          AND chk.checkWindow_id = @checkWindowId
+          AND chk.isLiveCheck = 1
+        ) latestPupilCheck
+        ON p.id = latestPupilCheck.pupil_id
+    LEFT JOIN [mtc_admin].[checkStatus] cs
+        ON (latestPupilCheck.checkStatus_id = cs.id)
+    LEFT JOIN [mtc_admin].pupilAttendance pa
+        ON (p.id = pa.pupil_id AND pa.isDeleted = 0)
+    LEFT JOIN [mtc_admin].attendanceCode ac
+        ON pa.attendanceCode_id = ac.id
```

Comment:
Do you think it would be worth combining these queries into a view?

The initial reason for breaking the queries up was to have two smaller queries hit the database rather than one large one.  This isn't relevant now we have a caching strategy.  However, a view would make support much easier.

Reply:
It does make sense. I'll update once this is done.

Reply:
Added